### PR TITLE
Support Rails 5

### PR DESCRIPTION
**Pull request by lorefnon**
Resolves Issue #385 

- [x] Configure test suite to run specs for active record 5
- [x] Identify rails 5 specific idiosyncracies in dependencies
- [x] Identify rails 5 specific idiosyncracies in core
- [x] Complete support for activerecord 3,4,5 and mongoid 3,4,5 

Suggestions and feedback are very welcome. 

tl;dr

The only primary issue with Rails 5 compatibility was that `belongs_to` is now required by default. In addition this PR configures the test suite to use [appraisal](https://github.com/thoughtbot/appraisal) to make it easier to support a range of ActiveRecord/Mongoid versions.

### CODE REVIEW

**Comment by wldcordeiro** [comment]
Thanks for doing this. I'll be looking in for review of the PR. If you need help just ping me.

**Comment by lorefnon** [comment]
I got occupied with other things and this PR got delayed. But I believe we can conclude this now. 

@wldcordeiro Please review at your convenience and let me know if any further changes are desired.

File: .travis.yml [hunk 1]
```diff
@@ -14,11 +8,80 @@ before_install:
...
 script: bundle exec rake
 
-env:
-  - ADAPTER=active_record
-  - ADAPTER=mongoid MONGOID_VERSION=3
-  - ADAPTER=mongoid MONGOID_VERSION=4
-  - ADAPTER=mongoid MONGOID_VERSION=5
+matrix:
+  include:
+    - rvm: 2.2.3
```
Comment (wldcordeiro) [inline .travis.yml]: Let's change this to test 2.2.x (where x is the latest release of the 2.2 line), 2.1.x and 2.0.x?

**Comment by wldcordeiro** [comment]
All good PR, I dig the addition of Appraisal. Just the one note but if you are too busy to update that file I can later. I'll be cutting a new release tomorrow with these changes.

**Comment by lorefnon** [comment]
@wldcordeiro Updated the Ruby versions as per your recommendation, the only exception being activerecord 5, which does not support 2.1.x and lower.

**Comment by wldcordeiro** [comment]
@lorefnon Makes sense. I'll be pulling in and (hopefully) getting a new release out today, just later after work. Thanks for the PR, the project is really community run now so if you ever wish to join the contributor team let me know. :+1:

**Comment by lorefnon** [comment]
@wldcordeiro Good to see this merged. I would be happy to be a part of the contributor team and help in whatever way I can.

**Pull request merged**